fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -256,7 +300,33 @@ public void setupScanRangeLocations() throws UserException {
             }
 
             StructLike partition = task.file().partition();
-            partitionKeyToId.putIfAbsent(partition, nextPartitionId());
+            long partitionId = 0;
+            if (!partitionKeyToId.containsKey(partition)) {
+                partitionId = nextPartitionId();
+                partitionKeyToId.put(partition, partitionId);
+                BiMap<Integer, PartitionField> indexToField = getIdentityPartitions(task.spec());
```

Comment:
Is `task.spec` same as it write instead of newest

Reply:
yes, i debug and confirmed it.